src/EnergyPlus/OutputProcessor.cc

```diff
@@ -4028,358 +3115,50 @@ namespace OutputProcessor {
-                         OutputProcessor::SOVStoreType const VariableTypeKey,    // State, Average=1, NonState, Sum=2
-                         std::string_view const KeyedValue,                      // Associated Key for this variable
-                         std::string_view const ReportFreq,                      // Internal use -- causes reporting at this frequency
-                         std::string_view const ResourceTypeKey,                 // Meter Resource Type (Electricity, Gas, etc)
-                         std::string_view const EndUseKey,                       // Meter End Use Key (Lights, Heating, Cooling, etc)
-                         std::string_view const EndUseSubKey,                    // Meter End Use Sub Key (General Lights, Task Lights, etc)
-                         std::string_view const GroupKey,                        // Meter Super Group Key (Building, System, Plant)
-                         std::string_view const ZoneKey,                         // Meter Zone Key (zone name)
-                         int const ZoneMult,                                     // Zone Multiplier, defaults to 1
-                         int const ZoneListMult,                                 // Zone List Multiplier, defaults to 1
-                         int const indexGroupKey,                                // Group identifier for SQL output
-                         std::string_view const customUnitName,                  // the custom name for the units from EMS definition of units
-                         std::string_view const SpaceType                        // Space type (applicable for Building group only)
+                         std::string_view const name,                  // String Name of variable (with units)
+                         Constant::Units const units,                  // Actual units corresponding to the actual variable
+                         Real64 &ActualVariable,                       // Actual Variable, used to set up pointer
+                         OutputProcessor::SOVTimeStepType sovTimeStep, // Zone, HeatBalance=1, HVAC, System, Plant=2
+                         OutputProcessor::SOVStoreType sovStore,       // State, Average=1, NonState, Sum=2
+                         std::string const &key,                       // Associated Key for this variable
+                         Constant::eResource resource,                 // Meter Resource Type (Electricity, Gas, etc)
+                         OutputProcessor::SOVEndUseCat sovEndUseCat,   // Meter End Use Key (Lights, Heating, Cooling, etc)
+                         std::string_view const EndUseSub,             // Meter End Use Sub Key (General Lights, Task Lights, etc)
+                         OutputProcessor::SOVGroup sovGroup,           // Meter Super Group Key (Building, System, Plant)
+                         std::string const &zone,                      // Meter Zone Key (zone name)
```

Comment:
One caution to point out here is that 

One example is the call example in PluginManager.cc (Lines 189-212):
The original way of call is still semantically valid; but would cause output problems due to the "zero" (or blank) style to place in parameters (e.g. the sovEndUseCat one):
```
SetupOutputVariable(state,
                                        sOutputVariable,
                                        thisUnit,
                                        sUpdateFreq,
                                        sAvgOrSum,
                                        thisObjectName,
                                        {},
                                        {}, // this one will be problematic
                                        {}, // this one will be problematic
                                        {},
                                        {}, // this one will be problematic
                                        {},
                                        1,
                                        1,
                                        -999,
```

Nice to see that the current code does correctly changed this to explicitly assigned "xxx:Invalide" to these few parameters:
```
SetupOutputVariable(state,
                                        sOutputVariable,
                                        thisUnit,
                                        sUpdateFreq,
                                        sAvgOrSum,
                                        thisObjectName,
                                        Constant::eResource::Invalid,
                                        OutputProcessor::SOVEndUseCat::Invalid,
                                        {}, // EndUseSub
                                        OutputProcessor::SOVGroup::Invalid,
                                        {}, // Zone
                                        1,
                                        1,
                                        -999,
```

Since the first style is semantically correct (and actually can commonly seen in the code in some  other function calls), a caution might be need to future developers (or the reviewers to be aware of this) that when calling this `SetupOutputVariable()` function, the `Constant::eResource resource`, `OutputProcessor::SOVEndUseCat sovEndUseCat`, and `OutputProcessor::SOVGroup sovGroup` needs to be explicitly put in as "xxx:Invalid" `enum` or some valid `enum` entry, if they are activated. They cannot be empty (zero) entries like `{},` or `,` which would cause (potentially big) output problems.

Or maybe there is a better way to better bullet-proof this?

Reply:
The way to bullet-proof this is to disallow this pattern by script. I should go back and replace `{}` with `""`.

Reply:
It took me sometime to get to figure this as well. However, here just replacing the curly bracket `{}` with `""` or `` (blank) might still not work---the `""` is an empty string and would even give some type mismatch error; the curly brackets and blank will send in a zero value which we require a `-1` (xxx:Invad `enum`) here. These few has to be explicitly the `enum` `xxx:Invalid` (`-1`) value. Maybe a `-1` might be working but I suspect it would trigger some (non-fatal) compiler warnings about parameter types.

Reply:
I meant just for string arguments.